src/components/PostContext.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { Post } from '../types/Post';
+import { getPosts } from '../utils/fetchClient';
+import { UserContext } from './UsersContext';
+
+type PostContextType = {
+  error: boolean;
+  setError: (error: boolean) => void;
+  errorNotification: string;
+  setErrorNotification: (error: string) => void;
+  posts: Post[];
+  setPosts: (posts: Post[]) => void;
+  postsLoading: boolean;
+  setPostsLoading: (postsLoading: boolean) => void;
+  selectedPost: Post | null;
+  setSelectedPost: (selectedPost: Post | null) => void;
+  showComments: boolean;
+  setShowComments: (showComments: boolean) => void;
+};
+
+export const PostContext = createContext<PostContextType>({
+  error: false,
+  setError: () => {},
+  errorNotification: '',
+  setErrorNotification: () => {},
+  posts: [] as Post[],
+  setPosts: () => {},
+  postsLoading: false,
+  setPostsLoading: () => {},
+  selectedPost: null,
+  setSelectedPost: () => {},
+  showComments: false,
+  setShowComments: () => {},
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const PostProvider = ({ children }: Props) => {
+  const [error, setError] = useState(false);
+  const [errorNotification, setErrorNotification] = useState('');
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [postsLoading, setPostsLoading] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [showComments, setShowComments] = useState(false);
+
+  const { selectedUser } = useContext(UserContext);
+
+  useEffect(() => {
+    setPostsLoading(true);
+    if (selectedUser) {
+      getPosts(selectedUser.id)
+        .then(setPosts)
+        .catch(() => {
+          setErrorNotification('Something went wrong');
+          setError(true);
+        })
+        .finally(() => {
+          setPostsLoading(false);
+        });
+    }
+  }, [selectedUser, setError]);
+
+  return (
+    <PostContext.Provider
+      value={{
+        error,
+        setError,
+        errorNotification,
+        setErrorNotification,
+        posts,
+        setPosts,
+        postsLoading,
+        setPostsLoading,
+        selectedPost,
+        setSelectedPost,
+        showComments,
+        setShowComments,
```

Comment:
the problem is that you're passing useState setters like setError and setPosts through context but they should stay in context because that's where they are created. To fix this you need to move all logic that uses them into context or create handler functions. Same is for all your contexts